micromanager_gui/main_window.py

```diff
@@ -148,92 +159,121 @@ def add_frame_explorer(self, name, array):
         except KeyError:
             self.viewer.add_image(array, name=layer_name)
 
-    # TO DO: add the file name form the save box
-    def _on_mda_frame(self, image, event):
 
-        sequence = event.sequence
+    # create temp folder when mda starts
+    def _on_mda_started_temp_folder(self, sequence: useq.MDASequence):
 
-        event.index.setdefault('t', 0)
-        event.index.setdefault('z', 0)
+        self.temp_folder = tempfile.TemporaryDirectory(None, str(sequence.uid))
+        return self.temp_folder.name
 
-        t_stack_length = len(sequence.time_plan) or 1
-        p_stack_length = len(sequence.stage_positions)
-        z_stack_length = len(sequence.z_plan) or 1
-        c_stack_length = len(sequence.channels)
+    #get the name to use to save the active ayer
+    def get_filename(self, fname, list_dir):
+        val = int(fname.split('_')[-1])
+        while True:
+            new_val = '{0:03}'.format(val)
+            fname = fname[:-3] + new_val
+            if ((fname + '.tif') or (fname + '.tiff')) not in set(list_dir):
+                break
+            else:
+                val += 1
+        return fname
 
-        event_index_t = event.index["t"]
-        event_index_p = event.index["p"]
-        event_index_z = event.index["z"]
-        event_index_c = event.index["c"]
-        
-        file_name = 'mda'
+    # delete temp folder and files when mda is done and save layer
+    # and add increment to save name
+    def _on_mda_finished_save_layer(self, sequence: useq.MDASequence):
+            
+        if self.mda.save_groupBox.isChecked():
```

Comment:
one problem here is that someone could have unchecked it in the meantime (and they wouldn't expect that it would "undo" saving the currently running experiment) ... so we might need to either store this on the sequence, or check for it in `_on_mda_started` and cache it for later

Reply:
can we just deactivate the possibility of checking it while the mda is running?

Reply:
is that better?  or just easier :)

Reply:
more to the point... it feels like a "patch".  It's definitely reasonable to disable some things while mda is running.  But I wouldn't _just_ disable the `save_groupBox` ... I would think more broadly about what should and shouldn't be interactive while mda is running and enable/disable _all_ of them.  for instance, perhaps the whole mda widget (besides the run/pause/cancel) buttons should be disabled.